main.tf

```diff
@@ -6,6 +6,10 @@ terraform {
       source  = "hashicorp/google"
       version = ">= 3.70.0"
     }
+    postgresql = {
+      source  = "cyrilgdn/postgresql"
+      version = ">= 1.11.0"
+    }
```

Comment:
Provider needs to be configured as well.

Reply:
Module configuration is inherited from the calling module: https://www.terraform.io/docs/language/modules/develop/composition.html

Reply:
Note that the `hashicorp/google` provider isn't configured either.

Reply:
For `google` provider that kinda makes sense, because its config requirements (the project) already exist.

However, for the `postgres` provider, the config requirements (the sql instance) are instantiated within this module. Going back and forth to the calling module feels kinda odd here. Especially since I wouldn't expect any interaction with the `postgres` provider *outside* of this module.

Reply:
Configuration of the provider is impossible, until the instance name and at least one administrative user is known, as those are needed to configure the provider.
Attempting to configure the provider in the module returns error, because resources are not yet known. In particular, in my attempt at setup, it says `values(google_sql_user.postgres_user)` is an empty list; while trying to fetch the first user created.
I also tried specifying a dependency in the provider config, but that resulted in this error:
> The provider argument name "depends_on" is reserved for use by Terraform in a future version.

Reply:
Taking this into account, I guess we should limit the module to privileged users (removing the `readonly` attribute for `var.users[]`) and manage readonly users outside the module.

Reply:
That feels even more icky. What about an instance module and an users module?